lua/terrortown/menus/gamemode/base_gamemodemenu.lua

```diff
@@ -102,6 +106,31 @@ function CLGAMEMODEMENU:AddSubmenu(submenu)
 	self.submenus[#self.submenus + 1] = submenu
 end
 
+---
+-- Checks if the menu has a searchbar enabled.
+-- @note This function should be overwritten and return true, if you want a searchbar.
+-- @return boolean Return true if searchbar should be available
+-- @realm client
+function CLGAMEMODEMENU:HasSearchbar()
+	return false
+end
+
+---
+-- Gets the used searchfunction.
+-- Parameters for that function are submenuClasses and the searchText
+-- @note This function can be overwritten to return a custom searchfunction.
+-- @return function Returns the default-function, which only searches the titles and compares strings in lowercase letters.
+-- @realm client
+function CLGAMEMODEMENU:GetSearchFunction()
+	return function(submenuClass, searchText)
+			local txt = stringLower(searchText)
+			local title = stringLower(TryT(submenuClass.title))
+			local start = stringFind(title, txt)
+
+			return start or false and true
+		end
+end
+
```

Comment:
So like this, I guess :) Unsure about the function indentation though.

Reply:
I still don't see the point here, sorry. Why not just:

```lua
function CLGAMEMODEMENU:GetSearchFunction(submenuClass, searchText)
	local txt = stringLower(searchText)
	local title = stringLower(TryT(submenuClass.title))
	local start = stringFind(title, txt)

	return start or false and true
end
```

Also your indentation is one too deep for our style

Reply:
Ahh, thats what you mean. It was my idea of localization. So you can just store that searchfunction. But I guess, the way you did it, I can still store it with submenuClass.SearchMenu ^^

Reply:
Obviously you spent much more time with the problem on hand. So if you think that your solution is the better one, I won't force you to change it. However with my insight based on this PR it seems to me as if your solution is unnecessary complicated

Reply:
No, I see your point. Im still stuck in a different mindset ^^
It's weird to be able to address function in a table directly, so I always feel like I want to give an object which is a function instead.